### PR TITLE
Add `modX` functions for all fields of `TxBodyContent` up to Babbage.

### PR DESCRIPTION
**Pull request by locallycompact**
# Changelog

```yaml
- description: |
    Adds `modTxTotalCollateral`, `modTxReturnCollateral`, `modTxFee`, `modTxValidityLowerBound`, `modTxValidityUpperBound`, `modTxMetadata`, `modTxAuxScripts`, `modTxWithdrawals`, `modTxCertificates`, `modTxUpdateProposal`, `modTxScriptValidity`, `modTxMintValue` functions for modifying `TxBodyContent`. Adds `addTxMintValue` and `subtractTxMintValue`.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/Tx/Body.hs [hunk 1]
```diff
@@ -1583,18 +1639,65 @@ setTxProtocolParams v txBodyContent = txBodyContent{txProtocolParams = v}
...
+  -> TxBodyContent build era
+  -> TxBodyContent build era
+addTxMintValue assets =
+  modTxMintValue
+    ( \case
+        TxMintNone -> TxMintValue maryBasedEra assets
+        TxMintValue era t -> TxMintValue era (t <> assets)
+    )
+
+subtractTxMintValue
```
Comment (smelc) [inline cardano-api/internal/Cardano/Api/Tx/Body.hs:1687]: Can you add documentation to this one? It seems to me one cannot guess what the implementation is doing just using the function name and the types involved.

**Review by smelc — APPROVED** [review]

**Comment by smelc** [comment]
@locallycompact> can I merge?

**Comment by locallycompact** [comment]
Yes please

**Comment by locallycompact** [comment]
Could we get a release with these in?

**Pull request merged**